zero/src/rpc/mod.rs

```diff
@@ -264,15 +263,25 @@ where
                         .into()
                 },
                 block_gas_used: target_block.header.gas_used.into(),
-                block_bloom: target_block.header.logs_bloom.compat(),
+                block_bloom: {
+                    const CHUNK: usize = 32;
+                    let Bloom(FixedBytes(bytes)) = target_block.header.logs_bloom;
+                    let chunks = bytes.chunks_exact(CHUNK);
+                    assert!(chunks.remainder().is_empty());
+                    let mut array = [U256::ZERO; 8];
+                    for (ix, chunk) in chunks.enumerate() {
+                        array[ix] = U256::from_le_bytes::<CHUNK>(chunk.try_into().unwrap());
+                    }
+                    array.map(alloy_compat::Compat::compat)
```

Comment:
Is this one worth its own function with generic parameters?

Reply:
Not yet I don't think - perhaps if it was used more than once?